nexus-sled-agent-shared/src/inventory.rs

```diff
@@ -591,6 +593,15 @@ impl ZoneKind {
     }
 }
 
+#[derive(
+    Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash,
+)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum OmicronZoneImageSource {
+    InstallDataset,
+    Artifact { hash: ArtifactHash },
+}
```

Comment:
This might be more of a question about the repo depot API - can sled-agent check that the `hash` provided here matches the zone type? E.g., I'd expect if I `PUT /omicron-zones` with a nonsense hash I'd get back an error that sled-agent has no such artifact, but ideally I'd also expect sled-agent to return an error if I tell it to run a Nexus zone with an artifact hash that points to an Oximeter zone image.

Reply:
Sled Agent doesn't currently know what type of artifact is what but I think it probably makes sense to either tell Sled Agent what the artifact types are (this could be added to the artifact config introduced in #7519) or have Sled Agent quickly inspect the artifact to make sure it looks like a zone (tar.gz, `oxide.json` is present with the correct type and perhaps even correct zone name).